esrally/driver/runner.py

```diff
@@ -520,15 +520,11 @@ def detailed_stats(self, params, response):
         total_document_size_bytes = 0
         with_action_metadata = mandatory(params, "action-metadata-present", self)
 
-        if isinstance(params["body"], str):
-            bulk_lines = params["body"].split("\n")
-        elif isinstance(params["body"], list):
-            bulk_lines = params["body"]
-        else:
-            raise exceptions.DataError("bulk body is neither string nor list")
+        if not isinstance(params["body"], bytes):
+            raise exceptions.DataError("bulk body is not of type bytes")
 
-        for line_number, data in enumerate(bulk_lines):
-            line_size = len(data.encode("utf-8"))
+        for line_number, data in enumerate(params["body"].split(b"\n")):
```

Comment:
This seems like a lighter thing to do, but since we are in the hot code path, did you check the performance impact between the two approaches?

Reply:
Is a microbenchmark enough?

```
∮ python -m timeit -s 's = "cafécafé\n".encode() * 1000' 's.split(b"\n")'
10000 loops, best of 5: 22.7 usec per loop
∮ python -m timeit -s 's = "cafécafé\n".encode() * 1000' 's.decode().split("\n")'
5000 loops, best of 5: 44.7 usec per loop
```

Note that we can´t really compare the previous approach because it would just raise an exception, as `params["body"]` became bytes at some point and `detailed-results` was thus broken until now.

Reply:
Yes ofc a microbenchmark for such cases is always very insightful. I haven't spent the time to go through all the places where bulk reads from a file, I was curious to see that we've removed the `data.encode('utf-8')` here, but added an `encode` in https://github.com/elastic/rally/pull/1400/files#diff-026a25af4075b17db2736adf14aa844c0aaeb2bc995b4b4ae94c432b0380b51fR350, hence wondering if we've measured the impact of the combined changes.

Reply:
This encode is in tests, though

Reply:
Ah yes ofc thank you! Sorry for the noise then and thanks for the microbenchmark.